infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batch/BatchResult.java

```diff
@@ -0,0 +1,40 @@
+package org.corfudb.infrastructure.log.statetransfer.batch;
+
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import static org.corfudb.infrastructure.log.statetransfer.batch.BatchResult.FailureStatus.SUCCEEDED;
+
+/**
+ * A result of a batch transfer. If completed successfully returns a list of addresses,
+ * status SUCCEEDED and an optional destination server.
+ * If completed exceptionally, returns a list of empty addresses,
+ * status FAILED and an optional destination server.
+ */
+@Builder
+@Getter
+@EqualsAndHashCode
+public class BatchResult {
+
+    public enum FailureStatus {
```

Comment:
The name `FailureStatus ` is confusing, maybe `TransferStatus`?

Reply:
Sounds good